src/use_case/display_profile/DisplayProfileInteractor.java

```diff
@@ -0,0 +1,39 @@
+package use_case.display_profile;
+
+import entity.User;
+import interface_adapter.display_profile.DisplayProfileState;
+import interface_adapter.display_profile.DisplayProfileViewModel;
+
+
+public class DisplayProfileInteractor implements DisplayProfileInputBoundary {
+    final DisplayProfileUserDataAccessInterface userDataAccessObject;
+    final DisplayProfileOutputBoundary displayProfilePresenter;
+    final DisplayProfileViewModel displayProfileViewModel;
+
+    public DisplayProfileInteractor(DisplayProfileUserDataAccessInterface userDataAccessInterface, DisplayProfileOutputBoundary displayProfileOutputBoundary
+    , DisplayProfileViewModel displayProfileViewModel) {
+        this.userDataAccessObject = userDataAccessInterface;
+        this.displayProfilePresenter = displayProfileOutputBoundary;
+        this.displayProfileViewModel = displayProfileViewModel;
+    }
+
+    @Override
+    public void execute(DisplayProfileInputData displayProfileInputData) {
+        String otherUsername = displayProfileInputData.getUsername();
+
+        User user = userDataAccessObject.get(otherUsername);
+
+        DisplayProfileState displayProfileState = displayProfileViewModel.getState();
```

Comment:
We shouldn't be calling the displayProfileState in the interactor as it belongs to the interface adapter layer. We probably don't need to pass the actual profile entity as the output data; we can just pass bio, artists, and Spotify handle separately.